var/spack/repos/builtin/packages/ip/package.py

```diff
@@ -16,17 +16,62 @@ class Ip(CMakePackage):
 
     maintainers("t-brown", "AlexanderRichert-NOAA", "edwardhartnett", "Hang-Lei-NOAA")
 
+    version("4.1.0", sha256="b83ca037d9a5ad3eb0fb1acfe665c38b51e01f6bd73ce9fb8bb2a14f5f63cdbe")
     version("4.0.0", sha256="a2ef0cc4e4012f9cb0389fab6097407f4c623eb49772d96eb80c44f804aa86b8")
     version(
         "3.3.3",
         sha256="d5a569ca7c8225a3ade64ef5cd68f3319bcd11f6f86eb3dba901d93842eb3633",
         preferred=True,
     )
 
+    variant("openmp", description="Enable OpenMP threading", default=True)
+    variant("pic", default=True, description="Build with position-independent-code")
+    variant("shared", default=False, description="Build shared library", when="@4.1:")
+    variant(
+        "precision",
+        default=["4", "d"],
+        values=["4", "d"],
+        multi=True,
+        description="Set precision (_4/_d library versions)",
+        when="@4.1:",
+    )
+    variant("tests", default=False, description="Build tests of the library")
```

Comment:
This shouldn't be a variant. Instead the corresponding logic should be enabled by `self.run_tests`. Or am I missing something?

Reply:
Is there some way to have a dependency conditioned on whether we're running a test? And for that matter, is there a way to capture the conflict with precision=4? If not then it may not be worth including the tests at the moment, in which case I can remove the variant and related logic.